### PR TITLE
ci(cross build): fix login error for aws ecr

### PR DESCRIPTION
**Pull request by Rory-Z**

### CODE REVIEW

**Review by zmstone — DISMISSED** [review]

File: .github/workflows/build_packages.yaml [hunk 1]
```diff
@@ -427,8 +427,7 @@ jobs:
     - uses: docker/build-push-action@v2
       with:
         push: >
-          ${{ (github.event_name == 'release' && !github.event.release.prerelease)
-          || (github.event.repository.owner != 'emqx' && startsWith(github.ref_name, 'ci/')) }}
```
Comment (qzhuyan) [inline .github/workflows/build_packages.yaml:431]: Could we keep the old one ? It is convenient for me to test on forked repo,  or doe it cause issue?

File: .github/workflows/build_packages.yaml [hunk 2]
```diff
@@ -563,17 +562,22 @@ jobs:
           scripts/docker-create-push-manifests.sh "${{ steps.meta.outputs.tags }}" "$IsPushLatest"
 
       - uses: aws-actions/configure-aws-credentials@v1
-        if: matrix.profile == 'emqx' && matrix.arch == 'amd64'
```
Comment (qzhuyan) [inline .github/workflows/build_packages.yaml:566]: keep 

if: matrix.arch == 'amd64'

File: .github/workflows/build_packages.yaml [hunk 3]
```diff
@@ -563,17 +562,22 @@ jobs:
...
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+        with:
+          registries: "public.ecr.aws"
+
       - name: Docker push to aws ecr
-        if: matrix.profile == 'emqx' && matrix.arch == 'amd64'
+        if: matrix.profile == 'emqx' && matrix.arch == 'amd64' && startsWith(github.ref, 'refs/tags/')
```
Comment (qzhuyan) [inline .github/workflows/build_packages.yaml:577]: Do you need to remove `matrix.profile == 'emqx'`?

File: .github/workflows/build_packages.yaml [hunk 4]
```diff
@@ -563,17 +562,22 @@ jobs:
...
 
       - uses: aws-actions/configure-aws-credentials@v1
-        if: matrix.profile == 'emqx' && matrix.arch == 'amd64'
         with:
           aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           aws-region: ${{ secrets.AWS_DEFAULT_REGION }}
+
+      - name: Login to Amazon ECR
+        id: login-ecr
```
Comment (qzhuyan) [inline .github/workflows/build_packages.yaml:571]: nit, add if conditions as the others.

File: .github/workflows/build_packages.yaml (hunk 1 above)
Comment (Rory-Z) [reply .github/workflows/build_packages.yaml:431]: Sure, I will restore it

File: .github/workflows/build_packages.yaml (hunk 3 above)
Comment (Rory-Z) [reply .github/workflows/build_packages.yaml:577]: No, we need `matrix.profile == 'emqx'`, because AWS ECR only has `emqx/emqx` register

**Review by zmstone — APPROVED** [review]

**Review by qzhuyan — APPROVED** [review]

**Pull request merged**